venues/IEEE.org/2018/ITSC/python/assign-reviewers.py

```diff
@@ -0,0 +1,122 @@
+parser.add_argument('--username')
+parser.add_argument('--password')
+args = parser.parse_args()
+
+## Initialize the client library with username and password
+client = Client(baseurl=args.baseurl, username=args.username, password=args.password)
+
+submissions = client.get_notes(invitation='IEEE.org/2018/ITSC/-/Submission')
+CONFERENCE_ID = 'IEEE.org/2018/ITSC'
+PROGRAM_CHAIRS = 'IEEE.org/2018/ITSC/Program_Chairs'
+## checks for valid paper numbers and email addresses
+def check_arguments(email_addr, paper_number):
+    try:
+        int(paper_number)
+    except ValueError:
+        print "Error: Paper number \"" + paper_number + "\" invalid."
+        return False
+
+    valid_email = re.compile('^[^@\s,]+@[^@\s,]+\.[^@\s,]+$')
+    if not valid_email.match(email_addr):
+        print "Error: Reviewer \""+email_addr+"\" invalid. Please check for typos and whitespace."
+        return False
+
+    return True
+
+# check if reviewer_email or reviewer's domains are on conflict list
+def reviewer_conflicts(reviewer_email, paper_number, conflict_list):
+    # check reviewer email as a whole is on the conflicts list
+    if reviewer_email in conflict_list:
+        print "Error: Reviewer \"" + reviewer_email + "\" is an author for Paper" + paper_number
+        return True
+
+    # if the reviewer email ends w/ .edu check it is on conflicts list
+    name, domain = reviewer_email.split('@')
+    if domain.endswith('.edu') and domain in conflict_list:
+        print "Error: Reviewer \"" + reviewer_email + "\" email has conflicts for Paper" + paper_number
+        return True
+
+    return False
+
+# does some parameter checking, checks for conflicts
+# check if reviewer already assigned to paper
+# if not then create Anonymous group for the reviewer and add to reviewer group
+def assign_reviewer(reviewer_email, paper_number):
+    if not check_arguments(reviewer_email, paper_number):
+        return False
+
+    # check paper_number exists
+    notes = [note for note in submissions if str(note.number) == paper_number]
+    if not notes:
+        print "Error: Paper number " + paper_number + " does not exist"
+        return True
+    note = notes[0]
+
+    # create list of conflicts emails and add paper author to it
+    # conflict_list is a copy (not a reference) to the submission conflicts
+    conflict_list = []
+    # If conflicts are a part of a submission, add them here
+    if 'conflicts' in note.content:
+        conflict_list.extend(note.content['conflicts'][:])
+    # authors for this paper are not allowed to be in the reviewers groups
+    if 'authorids' in note.content:
+        conflict_list += note.content['authorids']
+
+    if reviewer_conflicts(reviewer_email, paper_number, conflict_list):
+        return True
+
+    reviewer_id = tools.get_profile(client, reviewer_email)
+    if reviewer_id is None:
+        reviewer_id = reviewer_email
+    else:
+        reviewer_id = reviewer_id.id
```

Comment:
I made a PR on openreview-py earlier today that does this inside the tools.assign function. I'll leave this PR open for right now. If you have a chance to make the change then great, but this should work either way.